dev/requirements.txt

```diff
@@ -27,6 +27,7 @@ pytest-mypy-plugins==1.9.3
 flake8==3.9.0
 # See SPARK-38680.
 pandas-stubs<1.2.0.54
+scipy-stubs; python_version>='3.10'
```

Comment:
Thank you for making a PR, @jorenham .

May I ask why this is limited to Python 3.10+?

BTW, FYI, since Apache Spark 4.1.0 is going to drop Python 3.9, I believe we can remove this condition `python_version>='3.10'`.
- #51259

Reply:
cc @HyukjinKwon , @zhengruifeng , @xinrong-meng , @allisonwang-db from #51259

Reply:
> May I ask why this is limited to Python 3.10+?

at the earliest, `scipy-stubs` supports scipy `1.14`, which requires python >=3.10, as per [SPEC 0](https://scientific-python.org/specs/spec-0000/)